httpx/concurrency/asyncio.py

```diff
@@ -194,6 +194,44 @@ def loop(self) -> asyncio.AbstractEventLoop:
             stream_reader=stream_reader, stream_writer=stream_writer, timeout=timeout
         )
 
+    async def start_tls(
+        self,
+        stream: BaseStream,
+        hostname: str,
+        ssl_context: ssl.SSLContext,
+        timeout: TimeoutConfig,
+    ) -> BaseStream:
+
+        loop = self.loop
+        if not hasattr(loop, "start_tls"):  # pragma: no cover
+            raise NotImplementedError(
+                "asyncio.AbstractEventLoop.start_tls() is only available in Python 3.7+"
+            )
+
+        assert isinstance(stream, Stream)
+
+        stream_reader = asyncio.StreamReader()
+        protocol = asyncio.StreamReaderProtocol(stream_reader)
+        transport = stream.stream_writer.transport
+
+        loop_start_tls = loop.start_tls  # type: ignore
+        transport = await asyncio.wait_for(
+            loop_start_tls(
```

Comment:
Out of curiosity: why is it necessary to store `loop.start_tls` in a separate variable before using it here?

Reply:
Mypy complains about AbstractEventLoop not having start_tls() and I want type coverage for the rest of that line.

Reply:
Fine! 😄